docs/releasing.md

```diff
@@ -4,102 +4,40 @@ Describes how we do releases of Mithril.js
-4. Replace all existing references to `mithril@next` to `mithril` if moving from a release candidate to stable.
-    - Note: if making an initial release candidate, don't forget to move all the playground snippets to pull from `mithril@next`!
-5. Commit changes to `next`
-
-```
-$ git add .
-$ git commit -m "Preparing for release"
-
-# Push to your branch
-$ git push
-
-# Push to MithrilJS/mithril.js
-$ git push mithriljs next
-```
-
-### Merge from `next` to `master`
-
-6. Switch to `master` and make sure it's up to date
-
-```bash
-$ git checkout master
-$ git pull --rebase mithriljs master
-```
-
-7. merge `next` on top of it
-
-```bash
-$ git merge next
-```
+The description and title will be managed by [pr-release], including the semver version.
 
-8. Clean & update npm dependencies and ensure the tests are passing.
+Contributors who have permissions should add the correct semver label to their PR (`major` | `minor` | `patch`).  If no label is set, `patch` is assumed.
 
-```bash
-$ npm prune
-$ npm i
-$ npm test
-```
-
-### Publish the release
+If you do not have permissions, the maintainer will set the label on your behalf.
 
-9. `npm run release <major|minor|patch|semver>`, see the docs for [`npm version`](https://docs.npmjs.com/cli/version)
-10. The changes will be automatically pushed to your fork
-11. Push the changes to `MithrilJS/mithril.js`
-
-```bash
-$ git push mithriljs master
-```
+## Changelog
 
-12. Travis will push the new release to npm & create a GitHub release
+There are two changelogs in the Mithril.js project
 
-### Merge `master` back into `next`
+- `docs/changelog.md` a hand written curated reflection of changes to the codebase
+- `docs/release.md` an automatically prepended log of changes, managed by pr-release
 
-This helps to ensure that the `version` field of `package.json` doesn't get out of date.
+In future we may collapse these into a single file, the separation is due to the fact the `changelog.md` predates the `release.md` file.
 
-13. Switch to `next` and make sure it's up to date
+## For maintainers
 
-```bash
-$ git checkout next
-$ git pull --rebase mithriljs next
-```
+Whenever a new feature branch is opened, a reviewing maintainer should add the correct semver label to their PR (`major` | `minor` | `patch`).  If no label is set, `patch` is assumed.
 
-14. Merge `master` back onto `next`
-
-```bash
-$ git merge master
-```
-
-15. Push the changes to your fork & `MithrilJS/mithril.js`
-
-```bash
-$ git push
-$ git push mithriljs next
-```
-
-### Update the GitHub release
-
-16. The GitHub Release will require a manual description & title to be added. I suggest coming up with a fun title & then copying the `docs/changelog.md` entry for the build.
+If a `major` or `minor` feature branch is merge, but no labels were set, you can go back and apply the labels and then re-run the `pr` workflow in github actions, this will recalculate the semver version.
```

Comment:
Can we make the workflow [also trigger on label update](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#label)? Seems like an easy step to forget.